driver-core/src/main/com/mongodb/internal/connection/DefaultServerMonitor.java

```diff
@@ -79,11 +79,13 @@ class DefaultServerMonitor implements ServerMonitor {
     private final ServerApi serverApi;
     private final boolean faas;
     private final ServerSettings serverSettings;
-    private final ServerMonitorRunnable monitor;
-    private final Thread monitorThread;
-    private final RoundTripTimeRunnable roundTripTimeMonitor;
+    private final ServerMonitor monitor;
+    /**
+     * Must be guarded by {@link #lock}.
+     */
+    @Nullable
+    private RoundTripTimeMonitor roundTripTimeMonitor;
     private final ExponentiallyWeightedMovingAverage averageRoundTripTime = new ExponentiallyWeightedMovingAverage(0.2);
-    private final Thread roundTripTimeMonitorThread;
```

Comment:
The changes required both `roundTripTimeMonitor` & `roundTripTimeMonitorThread` to become `null`-able. I simplified that by replacing these two fields with a single field. However, this change lead to an inconsistent code having a single field `roundTripTimeMonitor` and two coupled fields `monitor` & `monitorThread`. To avoid such inconsistency, I also replaced the latter with a single field.

Reply:
Thank you, the code looks cleaner now 👍